membership_variable_period/models/product_template.py

```diff
@@ -63,6 +63,10 @@ def create(self, vals):
 
     @api.multi
     def write(self, vals):
-        self._correct_vals_membership_type(
-            vals, vals.get('membership_type', self.membership_type))
-        return super(ProductTemplate, self).write(vals)
+        if not vals.get('membership_type'):
+            return super(ProductTemplate, self).write(vals)
+        for rec in self:
+            rec._correct_vals_membership_type(
+                vals, vals.get('membership_type', rec.membership_type))
+            super(ProductTemplate, rec).write(vals)
```

Comment:
Agh, another side effect here: please make a copy of vals for each:
```suggestion
            vals2 = vals.copy()
            rec._correct_vals_membership_type(
                vals2, vals.get('membership_type', rec.membership_type))
            super(ProductTemplate, rec).write(vals2)
```